bilby/gw/waveform_generator.py

```diff
@@ -271,3 +272,243 @@ def validate_reference_frequency(self):
+                    pn_amplitude_order = 6  # Equivalent to MAX_NONPRECESSING_AMP_PN_ORDER in LALSimulation
+            start_frequency = minimum_frequency * 2. / (pn_amplitude_order + 2)
+        else:
+            start_frequency = minimum_frequency
+
+        params = self.defaults.copy()
+        params.update(parameters)
+        parameters = params
+
+        iota, spin_1x, spin_1y, spin_1z, spin_2x, spin_2y, spin_2z = bilby_to_lalsimulation_spins(
+            theta_jn=parameters["theta_jn"],
+            phi_jl=parameters["phi_jl"],
+            tilt_1=parameters["tilt_1"],
+            tilt_2=parameters["tilt_2"],
+            phi_12=parameters["phi_12"],
+            a_1=parameters["a_1"],
+            a_2=parameters["a_2"],
+            mass_1=parameters["mass_1"] * utils.solar_mass,
+            mass_2=parameters["mass_2"] * utils.solar_mass,
+            reference_frequency=reference_frequency,
+            phase=parameters["phase"],
+        )
+
+        longitude_ascending_nodes = 0.0
+
+        gwsignal_dict = {
+            'mass1': parameters["mass_1"],
+            'mass2': parameters["mass_2"],
+            'spin1x': spin_1x,
+            'spin1y': spin_1y,
+            'spin1z': spin_1z,
+            'spin2x': spin_2x,
+            'spin2y': spin_2y,
+            'spin2z': spin_2z,
+            'deltaF': 1 / self.duration,
+            'deltaT': 1 / self.sampling_frequency,
+            'f22_start': start_frequency,
+            'f_max': maximum_frequency,
+            'f22_ref': reference_frequency,
+            'phi_ref': parameters["phase"],
+            'distance': parameters["luminosity_distance"],
+            'inclination': iota,
+            'eccentricity': parameters["eccentricity"],
+            'longAscNodes': longitude_ascending_nodes,
+            'meanPerAno': parameters["mean_per_ano"],
+            'condition': self.condition
+        }
+        gwsignal_dict = {
+            key: val << Cosmo_units_dictionary.get(key, 0)
+            for key, val in gwsignal_dict.items()
+        }
+
+        if mode_array is not None:
+            gwsignal_dict.update(ModeArray=mode_array)
+
+        extra_args = waveform_kwargs.copy()
+
+        for key in [
+                "waveform_approximant",
+                "reference_frequency",
+                "minimum_frequency",
+                "maximum_frequency",
+                "catch_waveform_errors",
+                "mode_array",
+                "pn_spin_order",
+                "pn_amplitude_order",
+                "pn_tidal_order",
+                "pn_phase_order",
+                "numerical_relativity_file",
+        ]:
+            if key in extra_args.keys():
+                del extra_args[key]
+
+        gwsignal_dict.update(extra_args)
+        return gwsignal_dict
+
+    @property
+    def condition(self):
+        condition = 0
+        if self.generator.metadata["implemented_domain"] == 'time':
+            condition = 1
+        return condition
```

Comment:
Is there a reason to use `0/1` instead of `True/False`?

Reply:
The use of 0/1 was added in the first stage of the gwsignal development (see this [notebook](https://git.ligo.org/lscsoft/lalsuite/-/blob/master/lalsimulation/python/lalsimulation/gwsignal/docs/source/examples/testing_conditioning_routines.ipynb) for more details on the conditioning in gwsignal). Unfortunately, I do not know why 0/1 was chosen, instead of  False/True